taxi/tests/test_views.py

```diff
@@ -0,0 +1,436 @@
+        response = self.client.post(
+            reverse("taxi:manufacturer-delete", kwargs={"pk": manufacturer.pk})
+        )
+        self.assertEqual(response.status_code, 302)
+        self.assertRedirects(response, reverse("taxi:manufacturer-list"))
+        with self.assertRaises(manufacturer.DoesNotExist):
+            Manufacturer.objects.get(pk=manufacturer.pk)
+
+
+class PublicCarTests(TestCase):
+    def setUp(self) -> None:
+        model = "X5 M"
+        manufacturer = Manufacturer.objects.create(
+            name="TestName", country="TestCountry"
+        )
+
+        driver = Driver.objects.create_user(
+            username="TestUser",
+            password="TestPassword",
+            license_number="ABC12345"
+        )
+
+        car = Car.objects.create(
+            model=model,
+            manufacturer=manufacturer,
+        )
+
+        self.car = Car.objects.create(
+            model=model,
+            manufacturer=manufacturer,
+        )
+        car.drivers.add(driver)
+
+    def test_car_unauthorized_required_list(self) -> None:
+        response = self.client.get(CAR_LIST_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_car_unauthorized_required_create(self) -> None:
+        response = self.client.get(CAR_CREATE_URL)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_car_unauthorized_required_detail(self) -> None:
+        url = reverse("taxi:driver-detail", args=[self.car.pk])
+        response = self.client.get(url)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_car_unauthorized_required_update(self) -> None:
+        url = reverse("taxi:car-update", args=[self.car.pk])
+        response = self.client.get(url)
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_car_unauthorized_required_delete(self) -> None:
+        url = reverse("taxi:car-delete", args=[self.car.pk])
+        response = self.client.get(url)
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PrivateCarTests(TestCase):
+    @classmethod
+    def setUpTestData(cls) -> None:
+        manufacturer = Manufacturer.objects.create(
+            name="TestManufacturer", country="TestCountry"
+        )
+
+        driver = Driver.objects.create_user(
+            username="TestUser",
+            password="TestPassword",
+            license_number="ABC12345"
+        )
+
+        number_of_cars = 7
+        for car_id in range(number_of_cars):
+            car = Car.objects.create(
+                model=f"TestModel{car_id}", manufacturer=manufacturer
+            )
+            car.drivers.add(driver)
+
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            username="testuser", password="testpass"
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_car_list(self) -> None:
+        search_field_name = "model"
+        with self.assertNumQueries(4):

```

Comment:
The number of queries expected might need adjustment based on the actual queries executed during the request. Ensure that the number of queries matches the expected count.